odo/backends/sql.py

```diff
@@ -413,7 +413,7 @@ def append_select_statement_to_sql_Table(t, o, **kwargs):
     if not o.bind == t.bind:
         return append(t, convert(Iterator, o, **kwargs), **kwargs)
 
-    assert o.bind.has_table(t.name), 'tables must come from the same database'
+    assert o.bind.has_table(t.key), 'tables must come from the same database'
```

Comment:
So I thought that I would solve an issue I was having here by making this change; however, it did not work.

This assertion is catching false positives and I am not positive why.